scripts/auto-cite.py

```diff
@@ -0,0 +1,99 @@
+from util import *
+import importlib
+
+# config info for input/output files and plugins
+config = {}
+try:
+    config = load_data("./config.yaml", type_check=False)
+except Exception as message:
+    log(message, 3, "red")
+    exit(1)
+
+log("Compiling list of sources to cite")
+
+# compile master list of sources from various plugins
+sources = []
+
+# loop through plugins
+for plugin in config.get("plugins", []):
+    # get plugin props
+    name = plugin.get("name", "-")
```

Comment:
I like this new architecture! One last thing I might suggest would be to pass through all of the properties defined within the plugin's configuration section to the plugin itself. This way, a plugin could be configured either via a yaml file specific to the plugin or via a section in this file (it would be up to the plugin developer). This is looking great! 😁

Reply:
#73